src/patterns/ec2-app/base.ts

```diff
@@ -470,6 +511,46 @@ export class GuEc2App extends Construct {
       open: access.scope === AccessScope.PUBLIC,
     });
 
+    if (props.googleAuth?.enabled) {
+      const configPrefix = `${scope.stage}/${scope.stack}/${app}`;
+
+      const { clientIdPath = `/${configPrefix}/googleClientID`, clientSecretPath = `${configPrefix}/clientSecret` } =
+        props.googleAuth;
+
+      const clientId = StringParameter.fromStringParameterAttributes(this, "clientID", {
+        parameterName: clientIdPath,
+      }).stringValue;
+
+      // Unfortunately, Cloudformation doesn't support directly using secret
+      // Parameter Store values. But it is possible to use Secrets Manager.
+      const clientSecret = SecretValue.secretsManager(clientSecretPath);
+
+      const authAction = ListenerAction.authenticateOidc({
+        next: ListenerAction.forward([targetGroup]),
+        clientId: clientId,
+        clientSecret: clientSecret,
+        scope: "openid email",
+
+        // See the `hd` section of
+        // https://developers.google.com/identity/protocols/oauth2/openid-connect#authenticationuriparameters.
+        // Note, this is NOT sufficient to ensure access is limited to Guardian
+        // emails. Users should also ensure set 'User Type' to 'Internal' in the
+        // Google Console Consent Screen to ensure only @guardian.co.uk members
+        // can access your app and ensure (through Security Groups) that only
+        // the ALB is able to access the EC2 instances.
+        authenticationRequestExtraParams: { hd: "guardian.co.uk" },
```

Comment:
This allows anyone with a Guardian email address to authenticate. This is not enough to satisfy the requirements from @guardian/infosec, as we also need to validate one's membership in a Google Group.